File: atcoder/abc365a_leap_year.hs
-- vicfred
-- https://atcoder.jp/contests/abc365/tasks/abc365_a
-- implementation
main :: IO ()
main = getLine >>= print . numberOfDays . read

numberOfDays :: Int -> Int
numberOfDays year
            | year `mod` 4 == 0 && year `mod` 100 /= 0 = 366
            | year `mod` 100 == 0 && year `mod` 400 /= 0 = 365
            | year `mod` 400 == 0 = 366
            | otherwise = 365

File: atcoder/abc365a_leap_year.cpp
// vicfred
// https://atcoder.jp/contests/abc365/tasks/abc365_a
// implementation
#include <iostream>

using namespace std;

int main() {
  int Y;
  cin >> Y;
  if(Y % 4 != 0) {
    cout << 365 << endl;
  } else if(Y % 4 == 0 and Y % 100 != 0) {
    cout << 366 << endl;
  } else if(Y % 100 == 0 and Y % 400 != 0) {
    cout << 365 << endl;
  } else if (Y % 400 == 0) {
    cout << 366 << endl;
  }
  return 0;
}
